st2common/st2common/services/triggers.py

```diff
@@ -101,13 +101,22 @@ def get_trigger_type_db(ref):
 
 def _get_trigger_api_given_rule(rule):
     trigger = rule.trigger
-    triggertype_ref = ResourceReference.from_string_reference(trigger.get('type'))
     trigger_dict = {}
-
-    trigger_dict['name'] = triggertype_ref.name
-    trigger_dict['pack'] = triggertype_ref.pack
+    triggertype_ref = ResourceReference.from_string_reference(trigger.get('type'))
+    trigger_dict['pack'] = trigger_dict.get('pack', triggertype_ref.pack)
     trigger_dict['type'] = triggertype_ref.ref
-    trigger_dict['parameters'] = rule.trigger.get('parameters', {})
+
+    trigger_ref_str = trigger.get('ref', None)
```

Comment:
Yeah, I just tested it and it's broken again after this change.

`irc.pubmsg` trigger already exists in the DB, but when I create a new rule, another one with random name is created.

```
...
{ "_id" : ObjectId("54ba60f20640fd25ce7999f6"), "name" : "pubmsg", "pack" : "irc", "type" : "irc.pubmsg" }
....
{ "_id" : ObjectId("54bcdb040640fd0b8e5d87fa"), "name" : "e90fd53f-039e-4f17-a8b6-99ab71cfcc22", "pack" : "irc", "type" : "irc.pubmsg", "parameters" : {  } }
```

@lakshmi-kannan We should coordinate this on Slack.

As far as other "creating rule when trigger doesn't exist in db yet" issue goes - it seems like a simple solution would be to not allow rule creation if the referenced trigger doesn't exist in the DB yet (we should also return a friendly error message - e.g. "You need to register sensors/trigger before you can create rules which reference them").